components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/util/SAMLLogoutUtil.java

```diff
@@ -0,0 +1,307 @@
+        issuer.setValue(issuerID);
+        return issuer;
+    }
+
+    /**
+     * Validate the signature of the LogoutRequest against the given certificate.
+     *
+     * @param logoutRequest {@link LogoutRequest}object to be validated.
+     * @return true           If signature of the Logout Request is valid.
+     * @throws SAMLIdentityException Error when validating the signature.
+     */
+    public static boolean isValidSignature(LogoutRequest logoutRequest, SAMLMessageContext
+            samlMessageContext) throws SAMLIdentityException {
+
+        String issuer = logoutRequest.getIssuer().getValue();
+        X509Certificate x509Certificate = generateX509Certificate(samlMessageContext.
+                getFederatedIdP().getCertificate());
+
+        LogoutReqSignatureValidator signatureValidator = new LogoutReqSignatureValidator();
+        try {
+            if (samlMessageContext.getSAMLLogoutRequest().isPost()) {
+                return signatureValidator.validateXMLSignature(logoutRequest,
+                        new X509CredentialImpl(x509Certificate, issuer), null);
+            } else {
+                return signatureValidator.validateSignature(samlMessageContext.getSAMLLogoutRequest().getQueryString(),
+                        issuer, x509Certificate);
+            }
+        } catch (SecurityException | IdentityException e) {
+            throw new SAMLIdentityException("Process of validating the signature failed for the logout request with" +
+                    "issuer: " + logoutRequest.getIssuer().getValue(), e);
+        }
+    }
+
+    /**
+     * Generate the X509Certificate using the certificate string value in the identity provider's configuration.
+     *
+     * @param certificate String value of the certificate in the IdP's configurations.
+     * @throws SAMLIdentityException Error while generating the X509Certificate.
+     */
+    private static X509Certificate generateX509Certificate(String certificate)
+            throws SAMLIdentityException {
+
+        byte[] certificateData = java.util.Base64.getDecoder().decode(certificate);
+        try {
+            return (java.security.cert.X509Certificate) CertificateFactory.getInstance(CERTIFICATE_TYPE).
+                    generateCertificate(new ByteArrayInputStream(certificateData));
+        } catch (CertificateException e) {
+            throw new SAMLIdentityException("Error occurred while generating X509Certificate using the " +
+                    "string value of the certificate in IdP's properties: " + certificate, e);
+        }
+    }
+
+    /**
+     * @param logoutRequest {@link LogoutRequest} object.
+     * @return String               Session Index of the Logout Request.
+     * @throws SAMLIdentityException Error while extracting the Session Index.
+     */
+    public static String getSessionIndex(LogoutRequest logoutRequest) throws SAMLIdentityException {
+
+        if (CollectionUtils.isNotEmpty(logoutRequest.getSessionIndexes())) {
```

Comment:
assign `logoutRequest.getSessionIndexes()` to a list and resuse

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133